src/main/java/com/adobe/target/edge/client/ondevice/collator/CustomParamsCollator.java

```diff
@@ -35,4 +36,24 @@ private void addAllParameters(Map<String, Object> custom, RequestDetails details
       }
     }
   }
+
+  private Map<String, Object>  createNestedParametersFromDots(Map<String, Object> custom) {
+    Map<String, Object> result  = new HashMap<>();
+    custom.forEach((key, value) -> {
+      if (key.contains(".") & !key.contains("..")) {
```

Comment:
Did you mean to use the bitwise AND operator here?

https://www.freecodecamp.org/news/java-operator-and-or-logical-operators/

Reply:
Whoops, that was not my intention